### PR TITLE
Feature/client/t4 edit task

### PR DESCRIPTION
**Pull request by Atsuki02**
## Describe your changes

- create edit page
- Add tasks to the response of related houseApi and getMeApi 

## Issue

#39 

## URL
http://localhost:3000/tasks/:taskId/edit
(e.g. http://localhost:3000/tasks/126/edit)

## Screenshot (if you made the UI)

https://github.com/HAK2024/fairy-share/assets/132407687/f54a2437-559f-40e6-b41d-d78042100ad5

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings and no errors

### CODE REVIEW

File: server/src/house/house.service.ts
Comment (Atsuki02) [inline server/src/house/house.service.ts:1]: I added tasks to not only the response of getMeApi, but also that of houseApi to make it easier to handle the response type on frontend

File: client/app/tasks/[id]/edit/page.tsx [hunk 1]
```diff
@@ -1,5 +1,34 @@
...
+export default function EditTask({ params }: { params: { id: string } }) {
+  const taskId = Number(params.id)
+  const { isLoading: isLoadingHouse } = useGetHouseQuery()
+  const {
+    data: task,
+    isError: isTaskError,
+    isLoading: isLoadingTask,
+  } = useGetTaskQuery(taskId)
+
+  // If the user tries to access to the url that has invalid taskId, show an error component
```
Comment (kanatagu) [inline client/app/tasks/[id]/edit/page.tsx:18]: I think you don't need this comment?

File: client/app/tasks/hooks/useEditTask.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      date: defaultData.date,
+      assigneeId: defaultData.assigneeId,
+      note: defaultData.note,
+    },
+  })
+
+  const handleSuccess = () => {
+    toast({ variant: 'success', title: 'Successfully edited a task!' })
+    queryClient.invalidateQueries({
+      queryKey: ['todos', 'me', 'house', 'tasks'],
```
Comment (kanatagu) [inline client/app/tasks/hooks/useEditTask.ts:30]: Hmm. Do you also need to update me? I feel it's too much dependencies 🤔  Best practice is less dependencies... 
Btw, if you update one task, you want to just delete the case for one task Id!

File: client/app/tasks/schema/task.ts [hunk 3]
```diff
@@ -7,11 +7,18 @@ const taskSchema = z.object({
...
-  assigneeId: z.number({
-    required_error: 'Assignee is required',
-  }),
-
+  date: z
+    .date()
+    .nullable()
+    .refine((date) => date !== null, {
+      message: 'Date is required.',
+    }),
```
Comment (kanatagu) [inline client/app/tasks/schema/task.ts:15]: I got an error if you try to update without changing the date
Screenshot 2024-03-21 at 3 09 34 PM

File: client/app/tasks/[id]/edit/page.tsx (hunk 1 above)
Comment (Atsuki02) [reply client/app/tasks/[id]/edit/page.tsx]: Deleted it!
https://github.com/HAK2024/fairy-share/pull/84/commits/184a595460639b4f3543b67755cd993921f961c6

File: client/app/tasks/hooks/useEditTask.ts (hunk 2 above)
Comment (Atsuki02) [reply client/app/tasks/hooks/useEditTask.ts]: Fixed it!
https://github.com/HAK2024/fairy-share/pull/84/commits/0ad6a929330779aa97859f5327cc043d5ab0a2df

File: client/app/tasks/schema/task.ts (hunk 3 above)
Comment (Atsuki02) [reply client/app/tasks/schema/task.ts:15]: Fixed it!
https://github.com/HAK2024/fairy-share/pull/84/commits/b48a68a2011241d21f7df287e9b503538f38ba81

File: client/app/tasks/hooks/useEditTask.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+      note: defaultData.note,
+    },
+  })
+
+  const handleSuccess = () => {
+    toast({ variant: 'success', title: 'Successfully edited a task!' })
+    queryClient.invalidateQueries({
+      // TODO: Add the query key "tasks" when you need
+      queryKey: ['task', defaultData.id],
+    })
```
Comment (kanatagu) [inline client/app/tasks/hooks/useEditTask.ts:32]: I found out that you can just same key for tasks and task.id. 
```.js
// useGetTaskQuery
    queryKey: ['tasks', { id: taskId }],
```

```.js
    queryClient.invalidateQueries({
      // TODO: Add the query key "tasks" when you need
      queryKey: ['tasks', { id: defaultData.id }],
    })
```

https://tanstack.com/query/v4/docs/framework/react/guides/query-invalidation

**Review by kanatagu — APPROVED** [review]
LGTM!

File: client/app/tasks/hooks/useEditTask.ts (hunk 4 above)
Comment (Atsuki02) [reply client/app/tasks/hooks/useEditTask.ts]: Thanks! Fixed it!
https://github.com/HAK2024/fairy-share/pull/84/commits/d2e1b7191e4529b684796b264c74edc06e4476d9

**Pull request merged**